CONTRIBUTING.md

```diff
@@ -43,6 +43,13 @@ Here are some rules for editing the core BOUT++ code:
    add or update documentation where appropriate, we may ask you to do so before
    it is merged.
 
+-  As well as documentation for new features, you must also include a
+   representative test to ensure that it works correctly. Please see
+   the [tests README][test_readme] for more information on tests in
+   BOUT++. Prefer to write unit tests that check the feature at the function
+   level, rather than integrated tests that require setting up a whole physics
+   model.
+
```

Comment:
I agree with this: We don't want to discourage people from adding features, but also don't want to end up with unmaintainable code